pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/MailboxService.java

```diff
@@ -50,21 +55,51 @@
   int getMailboxPort();
 
   /**
-   * Look up a receiving mailbox by {@link MailboxIdentifier}.
-   *
-   * <p>the acquired {@link ReceivingMailbox} will be constructed if not exist already, but it might not have been
-   * initialized.
+   * Return a {@link ReceivingMailbox} for the given {@link MailboxIdentifier}.
    *
    * @param mailboxId mailbox identifier.
    * @return a receiving mailbox.
    */
   ReceivingMailbox<T> getReceivingMailbox(MailboxIdentifier mailboxId);
 
   /**
-   * Look up a sending mailbox by {@link MailboxIdentifier}.
+   * Same as {@link #getReceivingMailbox} but this would return null if the mailbox isn't already created.
+   */
+  @Nullable
+  ReceivingMailbox<T> getReceivingMailboxIfPresent(MailboxIdentifier mailboxId);
```

Comment:
This API is not needed. make it private and annotate the above method as Nullable.